k8s/upgrade/src/bin/upgrade-job/helm/yaml/yq.rs

```diff
@@ -169,9 +251,10 @@ impl YqV4 {
     }
 
     /// This sets in-place yaml values in yaml files.
-    pub(crate) fn set_value<V>(&self, key: YamlKey, value: V, filepath: &Path) -> Result<()>
+    pub(crate) fn set_literal_value<V, P>(&self, key: YamlKey, value: V, filepath: P) -> Result<()>
     where
         V: Display + Sized,
+        P: AsRef<Path>,
```

Comment:
why so many AsRef<Path> vs &Path?

Reply:
There were AsRef<Path>s and &Paths in this module. Made all of the args AsRef<Path>.

Reply:
kind of pointless here IMHO as paths returns &Path anyway, and need to add extra lines for P: and as_ref() throughout

Reply:
Alright, yeah that makes sense. These aren't standalone libraries anyway. Yup.